src/main/java/io/mantisrx/api/services/AppStreamDiscoveryService.java

```diff
@@ -15,85 +15,50 @@
  */
 package io.mantisrx.api.services;
 
-import io.mantisrx.shaded.com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.inject.Inject;
-import com.google.inject.name.Named;
-import com.netflix.config.DynamicStringProperty;
-import io.mantisrx.discovery.proto.AppJobClustersMap;
 import com.netflix.spectator.api.Counter;
 import com.netflix.zuul.netty.SpectatorUtils;
 import io.mantisrx.api.proto.AppDiscoveryMap;
 import io.mantisrx.client.MantisClient;
+import io.mantisrx.discovery.proto.AppJobClustersMap;
 import io.mantisrx.server.core.JobSchedulingInfo;
 import io.vavr.control.Either;
 import io.vavr.control.Option;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
 import lombok.extern.slf4j.Slf4j;
 import rx.Observable;
 import rx.Scheduler;
 
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicReference;
-
 @Slf4j
 public class AppStreamDiscoveryService {
 
-    private final AtomicReference<AppJobClustersMap> appJobClusterMappings = new AtomicReference<>();
-
     private final MantisClient mantisClient;
     private final Scheduler scheduler;
-    private final ObjectMapper objectMapper;
 
-    private final Counter appJobClusterMappingNullCount;
-    private final Counter appJobClusterMappingRequestCount;
-    private final Counter appJobClusterMappingFailCount;
+    private final AppStreamStore appStreamStore;
 
-    private final DynamicStringProperty appJobClustersProp = new DynamicStringProperty("mreAppJobClusterMap", "");
-
-    @Inject
-    public AppStreamDiscoveryService(MantisClient mantisClient,
-                                     @Named("io-scheduler") Scheduler scheduler,
-                                     ObjectMapper objectMapper) {
+    public AppStreamDiscoveryService(
+        MantisClient mantisClient,
+        Scheduler scheduler,
+        AppStreamStore appStreamStore) {
```

Comment:
I don't see the update on the new ctor passing in the instance of AppStreamStore?